update-docs.ps1

```diff
@@ -13,27 +13,33 @@
     PS C:\> ./update-docs.ps1 acronym
 #>
 
+[CmdletBinding(SupportsShouldProcess)]
 param (
     [Parameter(Position = 0, Mandatory = $false)]
     [string]$Exercise
 )
 
 # Import shared functionality
 . ./shared.ps1
+. ./update-canonical-data.ps1
 
-function Update-Canonical-Data {
-    Write-Output "Updating canonical data"
-    Run-Command "./update-canonical-data.ps1" 
-}
+function Update-DocsFolder {
+    [CmdletBinding(SupportsShouldProcess)]
+    param (
+        [Parameter(Position = 0, Mandatory = $false)]
+        [string]$Exercise
+    )
+
+    if ($PSCmdlet.ShouldProcess((& { If ($Exercise) { $Exercise } Else { "All Exercises" } }), "fetch configlet and pull specifications")) {
+        Write-Output "Updating docs"
+        Invoke-ExpressionExitOnError "./bin/fetch-configlet"
 
-function Update-Docs {
-    Write-Output "Updating docs"
-    $args = if ($Exercise) { @("-o", $Exercise) } else { @() }
-    Run-Command "./bin/fetch-configlet"
-    Run-Command "./bin/configlet generate . -p problem-specifications $args"
+        $configletArgs = if ($Exercise) { @("-e", $Exercise) } else { @() }
+        Invoke-ExpressionExitOnError "./bin/configlet sync -p problem-specifications -o $configletArgs"
+    }
 }
 
-Update-Canonical-Data
-Update-Docs
+Update-CanonicalData
+Update-DocsFolder $Exercise
```

Comment:
I see some changes where parameters are passed by name. Would it perhaps make sense to do that everywhere, as I think it can help with readability and also is beneficial when refactoring later on.

So something like:

```suggestion
Update-DocsFolder -Exercise $Exercise
```

And then require the parameter to be specified by name and not by position anymore.

Reply:
yes I did that.